src/core/Elsa.Core/Services/Workflows/ActivityActivator.cs

```diff
@@ -82,20 +87,31 @@ private async ValueTask ApplyStoredObjectValuesAsync(ActivityExecutionContext co
 
         private async ValueTask StoreAppliedValuesAsync(ActivityExecutionContext context, IActivity activity, CancellationToken cancellationToken)
         {
-            await StoreAppliedObjectValuesAsync(context, activity, cancellationToken);
+            await StoreAppliedObjectValuesAsync(context, activity, activity, cancellationToken);
         }
 
-        private async ValueTask StoreAppliedObjectValuesAsync(ActivityExecutionContext context, object activity, CancellationToken cancellationToken, string? parentName = null)
+        private async ValueTask StoreAppliedObjectValuesAsync(ActivityExecutionContext context, IActivity activity, object nestedInstance, CancellationToken cancellationToken, string? parentName = null)
```

Comment:
It's not really clear to me what the purpose of the nested instance is and the subsequent code changes. Perhaps this is a good candidate for some local comments explaining the logic.

Reply:
This actually comes from the nested properties, that are read from the activity's properties, that have the attribute `ActivityInputObjectAttribute`. The thing is though, that it seems this attribute is not even used anywhere. So I can only assume, that an activity can have properties with special types, that will recursively store their own properties, out of which there can be more such properties, which store their properties and so on.
What I am not sure is if such properties with that attribute are somehow supposed to an activity as the name in the arguments suggests, or is it something, that was forgotten to be renamed - to how I renamed it: `nestedInstance`.

Reply:
I see. It's a weird implementation, but not of your doing.